gateway-js/src/legacyLoadServicesFromStorage.ts

```diff
@@ -98,18 +98,22 @@ function fetchApolloGcs(
 }
 
 export async function getServiceDefinitionsFromStorage({
-  graphId,
+  graphRef,
   apiKeyHash,
-  graphVariant,
   federationVersion,
   fetcher,
 }: {
-  graphId: string;
+  graphRef: string;
   apiKeyHash: string;
-  graphVariant: string;
   federationVersion: number;
   fetcher: typeof fetch;
 }): Promise<ServiceDefinitionUpdate> {
+  // The protocol for talking to GCS requires us to split the graph ref
+  // into ID and variant; sigh.
+  const at = graphRef.indexOf('@');
+  const graphId = at === -1 ? graphRef : graphRef.substring(0, at);
+  const graphVariant = at === -1 ? 'current' : graphRef.substring(at + 1);
+
```

Comment:
```suggestion
  // The protocol for talking to GCS requires us to split the graph ref
  // into ID and variant; sigh.
  const [graphId, graphVariant = "current"] = graphRef.split("@");
```

Reply:
That's cool syntax I hadn't thought of. But if there are multiple `@`s it just silently drops whatever is after the second, right? Not really legal variant but still. (Super messed up that the limit arg to JS string.split isn't "stop splitting after N pieces" but "throw away all bit the first N pieces".)